src/daemon/globals.c

```diff
@@ -0,0 +1,13 @@
+#include "globals.h"
+#include "plugin.h"
+/*
+ * Global variables
+ */
+char hostname_g[DATA_MAX_NAME_LEN];
+const int hostname_g_size = sizeof (hostname_g);
+cdtime_t interval_g;
+int  pidfile_from_cli = 0;
+int  timeout_g;
+#if HAVE_LIBKSTAT
+kstat_ctl_t *kc;
```

Comment:
This type is not defined because the header is missing. I recommend to move

```c
#if HAVE_KSTAT_H
#include <kstat.h>
#endif
```

out of `src/daemon/collectd.h` and into here. Longer term, I'd like to get rid of the `kc` variable, but that's out of scope for here.